EIPS/eip-evm64-eof.md

```diff
@@ -0,0 +1,65 @@
+---
+title: EVM64 - EOF support
+description: EVM Object Format support for EVM64
+author: Wei Tang (@sorpaas)
+discussions-to: https://ethereum-magicians.org/t/eip-7937-64-bit-mode-evm-opcodes-evm64/23794
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-05-26
+requires: 3500, 3670, 4200, 7937
+---
+
+## Abstract
+
+This EIP defines EOF support for EVM64 with its additional code validation rules and `RJUMPI`, `RJUMPV` opcodes.
+
+## Motivation
+
+EOF defines a stricter stricter code validation rule to improve efficiency. Due to EVM64 using multibyte opcode (the mode opcode `C0`), a small adaptation is needed. This EIP also additionally defines a 64-bit mode `RJUMPI` and `RJUMPV` to be 64-bit.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+We define the following gas cost constant:
+
+* `G_RJUMPIV64`: 3
+
+If EOFv1 mode is entered, then an additional validation step is added. If the opcode `C0` is encountered and it is not part of PUSH opcode's data, then the interpreter MUST validate that:
```

Comment:
So this is validated at container deploy time, and not every time we enter an EOF contract? I assume this should be added to the validation logic of https://eips.ethereum.org/EIPS/eip-3670, but this is not entirely clear here